src/main/java/com/endava/Application.java

```diff
@@ -1,19 +1,55 @@
 package com.endava;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
 
 public class Application {
 
-    public static void main(String[] args) throws IOException {
+    private static final String PATH1 = "./src/main/resources/menu1.csv";
+    private static final String PATH2 = "./src/main/resources/menu2.csv";
+
+    public static void main(String[] args) throws IOException, ExecutionException, InterruptedException {
         double start = System.currentTimeMillis();
 
-        double menu1AveragePrice = FileUtil.getPriceAverage("./src/main/resources/menu1.csv");
-        double menu2AveragePrice = FileUtil.getPriceAverage("./src/main/resources/menu2.csv");
+        double menu1AveragePrice = FileUtil.getPriceAverage(PATH1);
+        double menu2AveragePrice = FileUtil.getPriceAverage(PATH2);
 
         double end = System.currentTimeMillis();
 
         System.out.println("Menu 1 average price: " + menu1AveragePrice);
         System.out.println("Menu 2 average price: " + menu2AveragePrice);
         System.out.println("Time to process: " + (end - start) / 1000 + " seconds");
+
+        start = System.currentTimeMillis();
+
+        List<Double> results = new ArrayList();
+
+        ExecutorService executorService = Executors.newFixedThreadPool(2);
+
+        Callable<List<Double>> callableTask = () -> {
+            double averagePrice1 = FileUtil.getPriceAverage(PATH1);
+            double averagePrice2 = FileUtil.getPriceAverage(PATH2);
+
+            results.add(averagePrice1);
+            results.add(averagePrice2);
+
+            return results;
+        };
+
+        Future<List<Double>> future = executorService.submit(callableTask);
```

Comment:
Same thing as Runnable ExecutorService PR, you need 2 Callables that you will submit to ExecutorService.